src/components/EventExpenses.tsx

```diff
@@ -0,0 +1,172 @@
+import { useMutation, useQuery } from "@apollo/client";
+import { useRouter } from "next/router";
+import { AiOutlineDelete } from "react-icons/ai";
+import { BiEdit } from "react-icons/bi";
+import client from "../configs/graphql";
+import { MemberRole } from "../constants";
+import { DELETE_EXPENSE } from "../graphql/mutations";
+import {
+  GET_EXPENSES_BY_CATEGORY,
+  GET_EXPENSES_OF_EVENT,
+} from "../graphql/queries";
+import PieChart from "./PieChart";
+import Table from "./Table";
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from "./ui/alert-dialog";
+import { useToast } from "./ui/use-toast";
+
+const EventExpenses = ({
+  eventID,
+  role,
+}: {
+  eventID: string;
+  role: MemberRole;
+}) => {
+  const router = useRouter();
+  const { data: expenses, loading } = useQuery(GET_EXPENSES_OF_EVENT, {
+    client: client,
+    fetchPolicy: "network-only",
+    variables: {
+      eventID,
+    },
+  });
+
+  const { data: expensesByCategory } = useQuery(GET_EXPENSES_BY_CATEGORY, {
+    client: client,
+    fetchPolicy: "network-only",
+    variables: {
+      eventID,
+    },
+  });
+  const categoriesWiseExpenses = expensesByCategory?.getExpensesByCategory;
+
+  const [deleteExpense] = useMutation(DELETE_EXPENSE, {
+    client: client,
+    refetchQueries: [GET_EXPENSES_OF_EVENT, "GetExpensesOfEvent"],
+  });
+  const { toast } = useToast();
+
+  const deleteExpenseHandler = async (id: string) => {
+    try {
+      const res = await deleteExpense({
+        variables: {
+          id,
+        },
+      });
+      if (res.data) {
+        toast({
+          title: "Success",
+          description: "Expense deleted successfully.",
+          variant: "success",
+        });
+      }
+    } catch (err: any) {
+      toast({
+        title: "Error",
+        description: err?.message,
+        variant: "destructive",
+      });
+    }
+  };
+
+  const editHandler = (data: any) => {
+    router.push({
+      pathname: `/dashboard/expense/add`,
+      query: {
+        event: eventID,
+        disable: true,
+        mode: "edit",
+        ...data,
+      },
```

Comment:
Better to keep the route path in a common constant  in structured format say, to avoid typo and repetition
```ts
const ROUTE_PATH = {
	DASHBOARD: {
		ROOT: '/dashboard',
		EXPENSE: {
			ROOT: '/dasboard/expense',
			ADD: '/dashboard/expense/add',
		}
	}
}
```

Also rather than keeping `eventID` in query param better to have it in URL path itself:
`/events/{eventID}` which makes `eventId` compulsory and avoids a scenario where the URL can be visited without it.